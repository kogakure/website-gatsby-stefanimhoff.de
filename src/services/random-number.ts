export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
