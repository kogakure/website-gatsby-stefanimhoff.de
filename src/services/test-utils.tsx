import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
