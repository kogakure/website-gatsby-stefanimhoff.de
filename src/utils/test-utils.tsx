import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
