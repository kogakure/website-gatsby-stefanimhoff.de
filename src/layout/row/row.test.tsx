import React from 'react';

import { render } from '../../services/test-utils';

import {
  BigLeft,
  BigRight,
  Center,
  Default,
  Equal,
  EqualCenter,
  EqualEnd,
  EqualStart,
  Fullsize,
  LeftDown,
  LeftOnly,
  RightDown,
  RightOnly,
  Variable,
  VariableCenter,
  VariableEnd,
  VariableStart,
} from './row.stories';

describe('Row', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('centered', () => {
    test('renders correctly', () => {
      const { container } = render(<Center />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('fullsize', () => {
    test('renders correctly', () => {
      const { container } = render(<Fullsize />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('only one item', () => {
    test('right renders correctly', () => {
      const { container } = render(<RightOnly />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('left renders correctly', () => {
      const { container } = render(<LeftOnly />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with big item', () => {
    test('right renders correctly', () => {
      const { container } = render(<BigRight />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('left renders correctly', () => {
      const { container } = render(<BigLeft />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with two equal sized items', () => {
    test('renders correctly', () => {
      const { container } = render(<Equal />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to start renders correctly', () => {
      const { container } = render(<EqualStart />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to center renders correctly', () => {
      const { container } = render(<EqualCenter />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to end renders correctly', () => {
      const { container } = render(<EqualEnd />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with one item down', () => {
    test('right renders correctly', () => {
      const { container } = render(<RightDown />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('left renders correctly', () => {
      const { container } = render(<LeftDown />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with variable amount of items', () => {
    test('renders correctly', () => {
      const { container } = render(<Variable />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to start renders correctly', () => {
      const { container } = render(<VariableStart />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to center renders correctly', () => {
      const { container } = render(<VariableCenter />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to end renders correctly', () => {
      const { container } = render(<VariableEnd />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
