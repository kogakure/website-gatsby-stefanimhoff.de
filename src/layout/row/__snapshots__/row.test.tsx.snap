// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Default
    </div>
  </div>
</section>
`;

exports[`Row with "bigLeft" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 1 / -6;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      bigLeft
    </div>
  </div>
</section>
`;

exports[`Row with "bigRight" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 6 / -1;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      bigRight
    </div>
  </div>
</section>
`;

exports[`Row with "center" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      center
    </div>
  </div>
</section>
`;

exports[`Row with "end" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      end
    </div>
    <div
      class="c3"
    >
      end
    </div>
  </div>
</section>
`;

exports[`Row with "fullSize" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      fullsize
    </div>
  </div>
</section>
`;

exports[`Row with "leftDown" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > *:first-child {
    margin-top: 11vw;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      leftDown
    </div>
    <div
      class="c2"
    >
      leftDown
    </div>
  </div>
</section>
`;

exports[`Row with "leftOnly" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > * {
    grid-column: 1;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      leftOnly
    </div>
  </div>
</section>
`;

exports[`Row with "middle" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      middle
    </div>
    <div
      class="c3"
    >
      middle
    </div>
  </div>
</section>
`;

exports[`Row with "rightDown" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > *:last-child {
    margin-top: 11vw;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      rightDown
    </div>
    <div
      class="c2"
    >
      rightDown
    </div>
  </div>
</section>
`;

exports[`Row with "rightOnly" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > * {
    grid-column: 2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      rightOnly
    </div>
  </div>
</section>
`;

exports[`Row with "start" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      start
    </div>
    <div
      class="c3"
    >
      start
    </div>
  </div>
</section>
`;

exports[`Row with "variableEnd" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  justify-items: center;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      variableEnd
    </div>
    <div
      class="c2"
    >
      variableEnd
    </div>
    <div
      class="c3"
    >
      variableEnd
    </div>
  </div>
</section>
`;

exports[`Row with "variableMiddle" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  justify-items: center;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      variableMiddle
    </div>
    <div
      class="c3"
    >
      variableMiddle
    </div>
    <div
      class="c2"
    >
      variableMiddle
    </div>
  </div>
</section>
`;

exports[`Row with "variableStart" renders correctly 1`] = `
.c0 {
  background-color: var(--colorForegroundFeather);
  border-color: green;
  border-style: dotted;
  border-width: 1px;
  width: 100%;
  margin-bottom: clamp(2.488rem,5.32vw,2.986rem);
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  justify-items: center;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(25,25,25,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      variableStart
    </div>
    <div
      class="c3"
    >
      variableStart
    </div>
    <div
      class="c3"
    >
      variableStart
    </div>
  </div>
</section>
`;
